relayer/crates/starknet-integration-tests/src/contexts/starknet_bootstrap.rs

```diff
@@ -172,6 +174,9 @@ impl ChainDriverBuilder<StarknetBootstrap> for BuildStarknetChainDriver {
 
         let rpc_client = Arc::new(JsonRpcClient::new(HttpTransport::new(json_rpc_url)));
 
+        // FIXME: `BuildAndWaitChainDriver`` doesn't help as this call doesn't wait for the chain to start
+        runtime.sleep(core::time::Duration::from_secs(5)).await;
+
```

Comment:
.. but I had to add a sleep inside `ChainDriverBuilder` impl as we are querying the RPC endpoint inside of it.

Reply:
That's because `BuildAndWaitChainDriver` waits on the chain driver _after_ it is constructed, but here we need to wait for it to call the underlying rpc_client _before_ it is constructed.

Here it is probably simpler to poll for the `chain_id()` in a loop, since when it succeeds the server also becomes ready.

Reply:
d762a09